05-mvc-with-rails/rails-views-and-controllers/readme.md

```diff
@@ -1,138 +1,325 @@
+
+```bash
+$ mkdir app/views/artists
+$ touch app/views/artists/index.html.erb
+```
+
+> Note the conventions here. We needed to make an `artists` folder to put the `index.html.erb` in it. This is important because when we define an `action` in our controller, rails knows to render the view corresponding to the controller and action. In this example, because were calling the `index` action in the `artists_controller`, it'll look for the `index` view in the `artists` folder.
+
+Inside `app/views/artist/index.html.erb`:
+Just put: `<h2>All Artists</h2>`
+
+Great, now let's refresh the page and there should be no more errors so we know everything has been wired up correctly!
+
+However, let's think about what we really want to see when we visit this page. When we visit `/artists`, we expect to see information about all artists!
+
+So now the question is how can we dynamically generate a view for all of our data about artists?
+
+### Instance Variables
+Recall from our Sinatra lessons how we used instance variables in our controllers.
+
+**Question:** What is the importance of instance variables in controller actions?
 
-## Instance variables (15/55)
 We use instance variables in our controller actions, so we can have programmatic access to variables inside of our views. More often than not these instance variables will contain objects from the database.
 
-Let's take our artists controller show action and add some things.
+Let's use our friend Active Record to query our database for all artists and save that to an instance variable.
+
+In the index action:
 ```ruby
-def show
-  @artist = Artist.find(params[:id])
-  @songs = @artist.songs
-  @food = "pizza"
+def index
+  @artists = Artist.all
 end
 ```
 
-We are looking for an artist in our database by it's id. We find it and store it in an instance variable. We can now use that inside the `app/views/artists/show.html.erb`
+Now, in our index view, we have programmatic access to the variable `@artists`!
 
 > It should be noted that while we practice a separation of concerns we aren't just limited to querying for only artists in the artists controller. Additionally, we can store just about anything we could have normally stored in a ruby variable.
 
-In our `app/views/artists/show.html.erb`
-```html
-<h1>My favorite food is <%= @food %></h1>
-<h2><%= @artist.name %> <a href="/artists/<%= @artist.id %>/edit">(edit)</a></h2>
-<h4><%= @artist.nationality %></h4>
+Let's write some code in our view to display this data.
 
-<img class='artist-photo' src="<%= @artist.photo_url %>">
+In our `app/views/artists/index.html.erb`
+```html
+<h2>Artists <a href="/artists/new">(+)</a></h2>
 
-<h3>Songs</h3>
 <ul>
-  <% @songs.each do |song| %>
-    <!-- build the list element for each song -->
+  <% @artists.each do |artist| %>
     <li>
-      <a href="/songs/<%= song.id %>">
-        <%= song.title %> (<%= song.album %>)
+      <a href="/artists/<%= artist.id %>">
```

Comment:
Why not `<%= link_to artist.name, artist %>`? Or are we saving that for the helpers class?